content/docs/atom-guide/dev-guide/shaders/azsl/_index.md

```diff
@@ -1,20 +1,468 @@
+* [The special attribute](#padton) [[pad_to(N)]].
+
+Although not a language extension, the shader build pipeline allows for *.azsl* files to include other files and define C/C++ Macros. The *.azsl* files are always pre-processed by the *C Pre-Processor (MCPP)* before being transpiled into HLSL. In other words, if a raw *.azsl* file has `#include`, `#pragma` or `#define` directives it should be run through **MCPP** before being compiled with **AZSLc**, otherwise **AZSLc** will fail to compile such files.
+
+**Caveat**: **MCPP**, doesn't support Macros with arithmetic expressions. Examples:  
+```cpp
+    // Bad (Arithmetic Expression in C-Macros are not supported by MCPP)
+    #define BAD_MACRO(x, y) ((x * 0.5) - (y * 2.0))
+     
+    // Good
+    #define HEIGHT (7.2)
+```
+## <a name="shaderconstants"></a>Declaring Shader Constants (also known as Shader Uniforms in other shader languages, like GLSL)
+
+AZSL only allows the declaration of shader constants inside the following places:
+
+* [Shader Resource Groups](#ShaderResourceGroup).
+* [Root Constants](#rootconstant).
+* `option` variables, but these variables can also be used equivalently as Vulkan's Specialization Constants, in other words, these constants can be statically defined at compilation time to avoid branching at runtime.  
+  
+**AZSLc** will trigger a compilation error if an arbitrary `cbuffer` or any other type of resources are declared outside the scopes mentioned above. The exception would be `static` global variables, which won't be exposed to the application but will be usable internally by the shaders.
+
+## <a name="ShaderResourceGroup"></a>Shader Resource Groups
+
+A *Shader Resource Group (SRG)* is a logical namespace to declare shader constants and resources. In AZSL the keyword to declare an SRG is `ShaderResourceGroup`. Shader resources are constants of the type Shader Resource Views (SRV), Unordered Access Views (UAV), Constant Buffer Views (CBV), and Sampler States.
+
+When compiling shaders with the "--use-spaces" argument, each SRG will take over a whole register space (or descriptor set, as known in Vulkan). AZSL abstracts the register and register space assignment of resource descriptors with the `ShaderResourceGroupSemantic` keyword, which must be specified when declaring SRGs.
+
+A Shader Resource Group is declared with the following syntax:
+  
+    ShaderResourceGroup <Name> : <Semantic>
+    {
+        <Data>
+    };
+     
+    //Alternatively
+    partial ShaderResourceGroup <Name> : <Semantic>
+    {
+        <Data>
+    };
+  
+* **&lt;Name&gt;** is the user defined name for the SRG.
+* **&lt;Semantic&gt;** is the name of the `ShaderResourceGroupSemantic` for the SRG. More on this later.
+* **&lt;Data&gt;** is one or more declaration of functions, constants and resources for the binding data contained in the SRG.  
+  
+In the code block shown above, a [partial](#partial) qualifier can be used when declaring SRGs.
+### &lt;Data&gt;
```

Comment:
```suggestion
### Data

SRGs can contain the following data: 
```
- Consider removing the < and > around Data, unless "<Data>" is the actual name of the code element. 
- Note the previous suggestion to reference this section. Update the link if you decide to name this section other than "Data". 
- Add a sentence to describe this section and what customers should do with the information. In this section you list the types of data that SRGs can contain. 
  - Consider using bullets to list these data types, to avoid repeating "SRGs can contain...". (See the following suggestions.)
  Example: 
  
  ![image](https://user-images.githubusercontent.com/75444793/143142754-c395aac0-c648-4a9c-86c7-d6e52378c803.png)

Reply:
Fixed.